action/protocol/staking/candidate.go

```diff
@@ -251,21 +252,36 @@ func delCandidate(sm protocol.StateManager, name address.Address) error {
 }
 
 func getAllCandidates(sr protocol.StateReader) (CandidateList, error) {
-	// TODO
-	return nil, nil
+	// TODO: load from current height's candidate center
+	_, iter, err := sr.States(protocol.NamespaceOption(factory.CandidateNameSpace))
+	if errors.Cause(err) == state.ErrStateNotExist {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var cands CandidateList
+	for i := 0; i < iter.Size(); i++ {
```

Comment:
you would prealloc slice here. `cands := make(CandidateList, 0, iter.Size())`

Reply:
ok